signalfx-yammer/src/main/java/com/signalfx/codahale/reporter/AggregateMetricSenderSessionWrapper.java

```diff
@@ -0,0 +1,273 @@
+package com.signalfx.codahale.reporter;
+
+import java.io.Closeable;
+import java.util.Map;
+import java.util.Set;
+
+import com.yammer.metrics.core.Histogram;
+import com.yammer.metrics.core.Metered;
+import com.yammer.metrics.core.Metric;
+import com.yammer.metrics.core.MetricName;
+import com.yammer.metrics.core.Sampling;
+import com.yammer.metrics.stats.Snapshot;
+import com.yammer.metrics.core.Timer;
+
+import com.google.common.base.Optional;
+import com.google.common.collect.ImmutableSet;
+import com.signalfx.metrics.SignalFxMetricsException;
+import com.signalfx.metrics.flush.AggregateMetricSender;
+import com.signalfx.metrics.protobuf.SignalFxProtocolBuffers;
+
+class AggregateMetricSenderSessionWrapper implements Closeable {
+    private final AggregateMetricSender.Session metricSenderSession;
+    private final Set<SignalFxReporter.MetricDetails> detailsToAdd;
+    private final MetricMetadata metricMetadata;
+    private final String defaultSourceName;
+    private final String sourceDimension;
+    private final boolean injectCurrentTimestamp;
+
+    /*AggregateMetricSenderSessionWrapper(
+            AggregateMetricSender.Session metricSenderSession,
+            Set<SignalFxReporter.MetricDetails> detailsToAdd,
+            MetricMetadata metricMetadata,
+            String defaultSourceName,
+            String sourceDimension) {
+        this(metricSenderSession, detailsToAdd, metricMetadata, defaultSourceName, sourceDimension, false);
+    }*/
+
+    AggregateMetricSenderSessionWrapper(
+            AggregateMetricSender.Session metricSenderSession,
+            Set<SignalFxReporter.MetricDetails> detailsToAdd,
+            MetricMetadata metricMetadata,
+            String defaultSourceName,
+            String sourceDimension,
+            boolean injectCurrentTimestamp) {
+        this.metricSenderSession = metricSenderSession;
+        this.detailsToAdd = detailsToAdd;
+        this.metricMetadata = metricMetadata;
+        this.defaultSourceName = defaultSourceName;
+        this.sourceDimension = sourceDimension;
+        this.injectCurrentTimestamp = injectCurrentTimestamp;
+    }
+
+    @Override
+    public void close() {
+        try {
+            metricSenderSession.close();
+        } catch (Exception e) {
+            throw new SignalFxMetricsException("Unable to close session and send metrics", e);
+        }
+    }
+
+    // These three called from report
+    void addTimer(MetricName key, Timer value) {
+        addMetered(key, value);
+        addSampling(key, value);
+    }
+
+    void addHistogram(MetricName baseName,
```

Comment:
Feel like this file looks almost exactly like https://github.com/signalfx/signalfx-java/blob/master/signalfx-codahale/src/main/java/com/signalfx/codahale/reporter/AggregateMetricSenderSessionWrapper.java

Can this not be refactored out to be common to both?